src/databricks/labs/ucx/hive_metastore/table_migrate.py

```diff
@@ -18,61 +19,49 @@ def __init__(
         tc: TablesCrawler,
         ws: WorkspaceClient,
         backend: SqlBackend,
-        default_catalog=None,
-        database_to_catalog_mapping: dict[str, str] | None = None,
+        tm: TableMapping,
     ):
         self._tc = tc
         self._backend = backend
         self._ws = ws
-        self._database_to_catalog_mapping = database_to_catalog_mapping
-        self._default_catalog = self._init_default_catalog(default_catalog)
+        self._tm = tm
         self._seen_tables: dict[str, str] = {}
 
-    @staticmethod
-    def _init_default_catalog(default_catalog):
-        if default_catalog:
-            return default_catalog
-        else:
-            return "ucx_default"  # TODO : Fetch current workspace name and append it to the default catalog.
-
     def migrate_tables(self):
         self._init_seen_tables()
+        mapping_rules = self._get_mapping_rules_dict()
         tasks = []
         for table in self._tc.snapshot():
-            target_catalog = self._default_catalog
-            if self._database_to_catalog_mapping:
-                target_catalog = self._database_to_catalog_mapping[table.database]
-            tasks.append(partial(self._migrate_table, target_catalog, table))
-        _, errors = Threads.gather("migrate tables", tasks)
-        if len(errors) > 0:
-            # TODO: https://github.com/databrickslabs/ucx/issues/406
-            # TODO: pick first X issues in the summary
-            msg = f"Detected {len(errors)} errors: {'. '.join(str(e) for e in errors)}"
-            raise ValueError(msg)
-
-    def _migrate_table(self, target_catalog: str, table: Table):
-        sql = table.uc_create_sql(target_catalog)
-        logger.debug(f"Migrating table {table.key} to using SQL query: {sql}")
-        target = f"{target_catalog}.{table.database}.{table.name}".lower()
-
-        if self._table_already_upgraded(target):
-            logger.info(f"Table {table.key} already upgraded to {self._seen_tables[target]}")
-        elif table.object_type == "MANAGED":
+            rule = mapping_rules.get(table.key)
+            if not rule:
+                logger.info(f"Skipping table {table.key} table doesn't exist in the mapping table.")
+                continue
+            tasks.append(partial(self._migrate_table, table, rule.as_uc_table_key))
+        Threads.strict("migrate tables", tasks)
+
+    def _migrate_table(self, src_table: Table, target_table_key):
+        sql = src_table.uc_create_sql(target_table_key)
+        logger.debug(f"Migrating table {src_table.key} to using SQL query: {sql}")
+
+        if self._table_already_upgraded(target_table_key):
+            logger.info(f"Table {src_table.key} already upgraded to {target_table_key}")
+            return True
+        if src_table.object_type == "MANAGED":
```

Comment:
```suggestion
        if src.object_type == "MANAGED":
           return self._migrate_managed(src, target_table_key)
        if src.kind == "VIEW":
          return self._migrate_view(src, target_table_key)
```

refactor these small things into separate methods in this PR

Reply:
This functionality is erroneous and will be rewritten in the next PR

Reply:
please do the method intro in this this PR, so that you change one method per PR

Reply:
Done